tests/test_components_pulses.py

```diff
@@ -0,0 +1,213 @@
+
+import pytest
+import numpy as np
+
+import extra
+
+from euxfel_bunch_pattern import PPL_BITS
+from extra.data import RunDirectory, SourceData, KeyData, by_id
+from extra.components import XrayPulses, OpticalLaserPulses
+
+
+def assert_equal_sourcedata(sd1, sd2):
+    assert isinstance(sd1, SourceData)
+    assert isinstance(sd2, SourceData)
+    assert sd1.source == sd2.source
+    assert sd1.train_ids == sd2.train_ids
+
+
+def assert_equal_keydata(kd1, kd2):
+    assert isinstance(kd1, KeyData)
+    assert isinstance(kd2, KeyData)
+    assert kd1.source == kd2.source
+    assert kd1.key == kd2.key
+    assert kd1.train_ids == kd2.train_ids
+
+
+def test_definitions():
+    XrayPulses.master_clock
+    XrayPulses.bunch_clock_divider
+    XrayPulses.bunch_repetition_rate
```

Comment:
Do these need asserts or something? Not sure what they're testing :thinking:

Reply:
What it does need is instantiation 🙈 

The test was meant as a smoke test those those attributes exist. Their values are in principle arbitrary and may even change, it's to have a central place for reference.

Reply:
Fixed in 5cf7a87